kube-client/src/api/remote_command.rs

```diff
@@ -135,30 +159,23 @@ impl AttachedProcess {
         if !self.has_stderr {
             return None;
         }
-
-        let mut state = self.state.lock().unwrap();
-        state.stderr_reader.take()
+        self.stderr_reader.take()
     }
-}
 
-impl Future for AttachedProcess {
-    type Output = Option<Status>;
+    /// Abort the background task, causing remote command to fail.
+    #[inline]
+    pub fn abort(&self) {
+        self.task.abort();
+    }
 
-    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
-        let mut state = self.state.lock().unwrap();
-        if state.finished {
-            Poll::Ready(state.status.take())
-        } else {
-            // Update waker if necessary
-            if let Some(waker) = &state.waker {
-                if waker.will_wake(cx.waker()) {
-                    return Poll::Pending;
-                }
-            }
+    /// Waits for the remote command task to complete.
+    pub async fn join(self) -> Result<(), Error> {
+        self.task.await.unwrap_or_else(|e| Err(Error::Spawn(e)))
+    }
 
-            state.waker = Some(cx.waker().clone());
-            Poll::Pending
-        }
+    /// Take a future that resolves with any status object or when the sender is dropped.
```

Comment:
Maybe a note here about this being expected to return a `Some` once since it's an added `.unwrap` expectation in user code.